bot/persistence/database_manager.py

```diff
@@ -25,30 +27,35 @@ def __init__(self, db_file: str):
         """
         self._db_file = db_file
         self.connection = None
+        self.is_in_memory = self._db_file == ':memory:'
 
     def __enter__(self):
         """Entry method of the context manager.
 
-        Opens the db connection using the `sqlite3` library.
+        Opens the db connection using the `sqlite3` library. If the database is an in-memory database and the connection has not yet been used, it will
+        establish one, otherwise reuse an existing one.
 
         Returns:
             Connection: The database connection object.
         """
-        try:
-            self.connection = sqlite3.connect(self._db_file)
-        except Error as error:
-            print(error)
+        if self.is_in_memory:
+            self.connection = get_in_memory_connection()
+        else:
+            try:
+                self.connection = sqlite3.connect(self._db_file)
+            except Error as error:
+                print(error)
```

Comment:
Instead of printing the error it would be better to use our logger so that it shows up in our log file.